examples/platforms/da15000/radio.c

```diff
@@ -318,32 +407,55 @@ ThreadError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, u
 
 void da15000RadioProcess(otInstance *aInstance)
 {
-    uint8_t EnableRX;
+    sTransmitStatus = kThreadError_None;
 
     if (sSendFrameDone)
     {
+        // Check FP bit in ACK response
+        if ((sTransmitFrame.mPsdu[0] & IEEE802154_ACK_REQUEST)     != 0 &&
+            (sReceiveFrame.mPsdu[0]  & IEEE802154_FRAME_TYPE_MASK) != 0)
+        {
+            sFramePending = ((sReceiveFrame.mPsdu[0] & IEEE802154_FRAME_PENDING) != 0);
+        }
+
         sRadioState = kStateReceive;
-        otPlatRadioTransmitDone(aInstance, &sTransmitFrame, false, kThreadError_None);
+        otPlatRadioTransmitDone(aInstance, &sTransmitFrame, sFramePending, sTransmitStatus);
+
+        sFramePending = false;
         sSendFrameDone = false;
     }
 
-    if (sSED)
+    if (sRadioState == kStateSleep && sGoSleep)
     {
-        if ((otPlatAlarmGetNow() - sSleepConstDelay) > 100)
-        {
-            EnableRX = 0;
-            FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &EnableRX);
-            ad_ftdf_sleep_when_possible(1);
-            sRadioState = kStateSleep;
-            sSED = false;
-        }
+        sGoSleep = false;
+
+        sEnableRX = 0;
+        FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &sEnableRX);
+        ad_ftdf_sleep_when_possible(true);
     }
 }
 
 void FTDF_sendFrameTransparentConfirm(void *handle, FTDF_Bitmap32 status)
 {
     (void)handle;
-    (void)status;
+
+    if (status == FTDF_TRANSPARENT_SEND_SUCCESSFUL)
```

Comment:
This string of if/else-if statements looks like it may be better suited to be a switch case.